docker/ci/dockerfiles/current/build.al2.opensearch.x64.arm64.dockerfile

```diff
@@ -82,14 +82,14 @@ ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib:/usr/local/lib64:/usr/lib
 RUN yum install -y curl libcurl-devel libfaketime perl-core pcre-devel && yum remove -y openssl-devel && yum clean all && \
     mkdir -p /tmp/openssl && cd /tmp/openssl && \
     curl -sSL -o- https://www.openssl.org/source/openssl-1.1.1g.tar.gz | tar -xz --strip-components 1 && \
-    ./config --prefix=/usr --openssldir=/etc/ssl --libdir=lib shared zlib-dynamic && make && make install && \
+    ./config --prefix=/usr --openssldir=/etc/ssl --libdir=lib shared zlib-dynamic && make -j$(nproc) && make install && \
```

Comment:
my only concern is with using all the cores to parallelize the build. Wouldn't it slow down other build or processes running on the host if consuming all the cores?
Can we limit it to half or hardcode to 4 or 8?

Reply:
It wont, because by definition docker builder ubuntu2004 has one and only one executor.
https://github.com/opensearch-project/opensearch-ci/blob/main/lib/compute/agent-nodes.ts#L169

So we good.